source/common/ur_util.hpp

```diff
@@ -295,4 +295,12 @@ inline ur_result_t exceptionToResult(std::exception_ptr eptr) {
 
 template <class> inline constexpr bool ur_always_false_t = false;
 
+[[noreturn]] inline void ur_unreachable() {
+#ifdef _MSC_VER
+    __assume(0);
+#else
+    __builtin_unreachable();
+#endif
+}
```

Comment:
I think we should put this in a namespace `ur::unreachable()`.

Reply:
done